midp/fs.js

```diff
@@ -267,375 +263,463 @@ Native["com/sun/midp/rms/RecordStoreRegistry.stopAllRecordStoreListeners.(I)V"]
     console.warn("RecordStoreRegistry.stopAllRecordStoreListeners.(I)V not implemented (" + taskId + ")");
 };
 
-Native["com/ibm/oti/connection/file/Connection.isValidFilenameImpl.([B)Z"] = function(path) {
-    var invalid = ['<', '>', ':', '"', '/', '\\', '|', '*', '?'].map(function(char) {
-      return char.charCodeAt(0);
-    });
+Native["com/sun/midp/io/j2me/storage/RandomAccessStream.open.(Ljava/lang/String;I)I"] = function(fileName, mode) {
+    var path = "/" + util.fromJavaString(fileName);
 
-    for (var i = 0; i < path.length; i++) {
-        if (path[i] <= 31 || invalid.indexOf(path[i]) != -1) {
-            return 0;
+    var ctx = $.ctx;
+    asyncImpl("I", new Promise(function(resolve, reject) {
+        function open() {
+            fs.open(path, function(fd) {
+                ctx.setAsCurrentContext();
```

Comment:
> No need to setAsCurrentContext here, you could either setAsCurrentContext in the reject branch, or simply use ctx.runtime.newIOException(

Erm, I haven't actually touched this code. Git's default diff algorithm is just making it look like I did. I moved the new natives to the location of the old natives to make it more obvious which code I'm replacing.